composer/trainer/trainer.py

```diff
@@ -187,38 +191,26 @@ def __init__(
         seed_all(seed)
         self.seed = seed
 
-        if not algorithms:
-            algorithms = []
-
-        self.backwards_create_graph = any(map(lambda x: x.backwards_create_graph, algorithms))
-
-        find_unused_parameters = any(map(lambda x: x.find_unused_parameters, algorithms))
-
-        self.find_unused_parameters = find_unused_parameters
-
         if self.deepspeed_enabled:
             import deepspeed
             deepspeed.init_distributed()
         else:
             ddp.initialize_ddp(device.ddp_backend, datetime.timedelta(seconds=ddp_timeout))
-            if ddp_sync_strategy is None:
-                self.ddp_sync_strategy = ddp.DDPSyncStrategy.SINGLE_AUTO_SYNC if not find_unused_parameters else ddp.DDPSyncStrategy.FORCED_SYNC
-            else:
-                self.ddp_sync_strategy = ddp.DDPSyncStrategy(ddp_sync_strategy)
 
         if isinstance(train_dataloader, DataloaderSpec):
-            train_dataloader_spec = train_dataloader
+            self._train_device_transformation_fn = train_dataloader.device_transform_fn
```

Comment:
These changes don't seem related to binding the trainer arguments as properties?

I also find the previous logic to be more readable, and now the handling of train_dataloader and eval_dataloader are not consistent.